engine/src/main/java/org/hibernate/search/bridge/spi/FieldType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.spi;
+
+/**
+ * The type of an indexed field.
+ *
+ * @author Gunnar Morling
+ * @hsearch.experimental This contract is currently under active development and may be altered in future releases,
+ * breaking existing users.
+ */
+public enum FieldType {
+
+	/**
+	 * A String field.
+	 */
+	STRING,
+
+	/**
+	 * A boolean field, mapped to the String "true" or "false" in Lucene.
+	 */
+	STRING_MAPPED_BOOLEAN,
+
+	/**
+	 * A date, mapped to a String in Lucene.
+	 */
+	STRING_MAPPED_DATE,
+
+	/**
+	 * An integer field.
+	 */
+	INTEGER,
```

Comment:
We should make that clear, as the user needs to make sure the choice is consistent with the writing.
Or we evolve this further and take the different writing methods away from him, so to only have the choice from the metadata available ;) Not sure if we can though, we'd need to accept an Object type for the value.